### ISSUE TITLE
Add ability to consume from frozen_mutation in reverse

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylladb#11199]
Depending on #11198, we should extend `frozen_mutation::consume` and `frozen_mutation::consume_gently` to consume in reverse as we do in their `mutation::consume*` counterparts.

Otherwise, we currently defer (inefficiently) to unfreezing it and consuming the thawed mutation - see https://github.com/scylladb/scylladb/blob/a4844826fcc7a845a03a1c927a5bfe3d57c09901/mutation_partition.cc#L2214-L2226

**Comment by bhalevy** [comment]
@enedil let's start working on this issue.

**Comment by bhalevy** [comment]
Closing in favor of https://github.com/scylladb/scylladb/issues/12557.
